Debezium extensions native compilation is broken

### Bug description

This extension seem to work:

* `camel-quarkus-debezium-oracle`

These do not work:

* `camel-quarkus-debezium-mongodb`
* `camel-quarkus-debezium-mysql`
* `camel-quarkus-debezium-postgres`
* `camel-quarkus-debezium-sqlserver`

```
Caused by: java.lang.ClassNotFoundException: org.apache.maven.artifact.versioning.InvalidVersionSpecificationException
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageClassLoader.loadClass(NativeImageClassLoader.java:637)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
	... 63 more
```

```
Internal exception: com.oracle.svm.core.util.UserError$UserException: Discovered unresolved method during parsing: org.apache.kafka.connect.runtime.AbstractHerder.generateResult(java.lang.String, java.util.Map, java.util.List, java.util.List). This error is reported at image build time because class io.debezium.embedded.async.AsyncEmbeddedEngine is registered for linking at image build time by command line and command line.
Error encountered while parsing io.debezium.embedded.async.AsyncEmbeddedEngine.initializeConnector(AsyncEmbeddedEngine.java:365)
Parsing context:
   at io.debezium.embedded.async.AsyncEmbeddedEngine.run(AsyncEmbeddedEngine.java:208)
   at io.quarkus.runtime.StartupContext.runAllAndClear(StartupContext.java:79)
   at io.quarkus.runtime.StartupContext.close(StartupContext.java:70)
   at io.quarkus.runner.ApplicationImpl.doStop(Unknown Source)
   at io.quarkus.runtime.Application.stop(Application.java:223)
   at io.quarkus.runtime.Application.close(Application.java:152)
   at org.apache.camel.util.IOHelper.closeWithException(IOHelper.java:440)
   at org.apache.camel.util.IOHelper.closeIterator(IOHelper.java:493)
   at org.apache.camel.support.GroupIterator.close(GroupIterator.java:60)
   at org.apache.camel.util.IOHelper.close(IOHelper.java:418)
   at org.apache.camel.util.IOHelper.copyAndCloseInput(IOHelper.java:281)
   at org.apache.camel.converter.IOConverter.toBytes(IOConverter.java:291)
   at org.apache.camel.converter.CamelBaseBulkConverterLoader.doConvertTo(CamelBaseBulkConverterLoader.java:89)
   at org.apache.camel.converter.CamelBaseBulkConverterLoader.convertTo(CamelBaseBulkConverterLoader.java:56)
   at org.apache.camel.spi.BulkTypeConverters.tryConvertTo(BulkTypeConverters.java:70)
   at org.apache.camel.spi.BulkTypeConverters.tryConvertTo(BulkTypeConverters.java:150)
   at org.apache.camel.impl.converter.CoreTypeConverterRegistry.doConvert(CoreTypeConverterRegistry.java:477)
   at static root method.(Unknown Source)
```

When you add `camel-quarkus-debezium-oracle` along with any of the other extensions, it magically fixes things. That probably explains why we have not detected any problems in CI, because the test groups all Debezium extensions together.

--- a/extensions-support/debezium/deployment/pom.xml
+++ b/extensions-support/debezium/deployment/pom.xml
@@ -41,12 +41,6 @@
         <dependency>
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-support-debezium</artifactId>
-            <exclusions>
-                <exclusion>
-                    <groupId>org.apache.maven</groupId>
-                    <artifactId>maven-artifact</artifactId>
-                </exclusion>
-            </exclusions>
         </dependency>
     </dependencies>
 

